app/models/subnet_domain.rb

```diff
@@ -2,9 +2,6 @@ class SubnetDomain < ActiveRecord::Base
   belongs_to :domain
   belongs_to :subnet
 
-  validates :subnet_id, :presence => true
-  validates :domain_id, :presence => true, :uniqueness => {:scope => :subnet_id}
```

Comment:
This validates internal data that are not provided by user. Active record already catches invalid ids so we don't need to validate anything. What we really need is DB constraint, I'll add the migration to make both fields required, I was not aware it's missing as well. Makes sense?